sst/test/include/framework/parser.hh

```diff
@@ -0,0 +1,119 @@
+#pragma once
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+struct ParsedMemRead {
+  uint64_t reqId_;
+  uint64_t startCycle_;
+  uint64_t endCycle_;
+  uint64_t data_;
+  uint64_t numReqs_;
+};
+
+/**
+ * Parser class used to parse captured stdout into meaninful and comparable
+ * data.
+ */
+class Parser {
+ private:
+  /** Captured Stdout split into lines. */
+  std::vector<std::string> splitStdout_;
+
+  /** vector of ParsedMemReads. */
+  std::vector<ParsedMemRead*> parsedMemReads_;
+
+  /** vector of stdout lines having the SimEng:SSTDebug:OutputLine prefix. */
+  std::vector<std::string> outputLines;
+
+ public:
+  Parser(std::string capturedStdOut) {
+    splitStdout_ = splitStdoutIntoLines(capturedStdOut);
+    parsedMemReads_ = parseOutput();
+  };
+  ~Parser(){};
+  /** Returns the parsed stdout as ParsedMemReads. */
+  std::vector<ParsedMemRead*> getParsedMemReads() { return parsedMemReads_; }
+
+  /**
+   * Returns a vector of stdout lines having the SimEng:SSTDebug:OutputLine
+   * prefix.
+   */
+  std::vector<std::string> getOutputLines() { return outputLines; }
+
+ private:
+  /** This methiod parses the captured stdout */
+  std::vector<ParsedMemRead*> parseOutput() {
+    std::map<uint64_t, ParsedMemRead*> pmap;
+    std::vector<ParsedMemRead*> out;
+    for (size_t x = 0; x < splitStdout_.size(); x++) {
+      if (splitStdout_[x].find("SimEng:SSTDebug") == std::string::npos)
+        continue;
+      std::vector<std::string> splitStr = split(splitStdout_[x], "-");
+      if (splitStr[0] == "[SimEng:SSTDebug:MemRead]") {
+        uint64_t id = std::stoull(splitStr[3]);
+        auto itr = pmap.find(id);
+        if (itr != pmap.end()) {
+          itr->second->endCycle_ = std::stoull(splitStr[5]);
+          itr->second->data_ = std::stoull(splitStr[7]);
+          out.push_back(itr->second);
+        } else {
+          ParsedMemRead* p = new ParsedMemRead();
+          p->reqId_ = id;
+          p->startCycle_ = std::stoull(splitStr[5]);
+          p->numReqs_ = std::stoull(splitStr[7]);
+          pmap.insert(std::pair<uint64_t, ParsedMemRead*>(id, p));
+        }
+      }
+      if (splitStr[0] == "[SimEng:SSTDebug:OutputLine]") {
```

Comment:
Unsure what was decided in the end regarding the fidelity of the output prefix. I think it was [SimEng:LowestLevelComponent] rather than [SimEng:component:foo:bar] etc.

@jj16791 input on this would be appreciated

Reply:
For this, I think it would be best as just `[SimEng:SSTDebug]` for the sake of consistency

Reply:
There are various types of parsed outputs i.e `[SimEng::SSTDebug:MemRead]` and `[SimEng::SSTDebug:OutputLine]`. The first one parses information related to a read memory request into struct, the second records the entire line into string vector. Happy to replace the second `:` with a hypen i.e `[SimEng:SSTDebug-MemRead]` and [SimEng:SSTDebug-OutputLine]`

Reply:
I think the prefix should be describing the file / component producing the message, and output types such as "OutputLine" or "MemRead" should be part of the main message body. Keeps the consistency of prefixes without making them bloated, and keeps the output descriptive

Reply:
I second 

> I think the prefix should be describing the file / component producing the message, and output types such as "OutputLine" or "MemRead" should be part of the main message body. Keeps the consistency of prefixes without making them bloated, and keeps the output descriptive

I second this. It should just be `[SimEng:SSTDebug]` and any additional information should be encoded within the remainder of the output. It's then the job of the parser to dead with each message differently.

Reply:
This has been fixed. Please check.